tests/unit/test_gce_authority.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import contextlib
+import logging
+
+import pytest
+
+from gordon_janitor_gcp import exceptions
+from gordon_janitor_gcp import gce_authority
+from gordon_janitor_gcp import gcp_clients
+from gordon_janitor_gcp import msg_generators
+
+
+def echoing_helper_coro(data):
+    async def _coro():
+        return data
+    return _coro()
+
+
+@pytest.fixture
+def authority_config():
+    return {
+        'crm_keyfile': 'crm-keyfile',
+        'crm_scopes': ['crm-scope'],
+        'gce_keyfile': 'gce-keyfile',
+        'gce_scopes': ['gce-scope'],
+        'metadata_blacklist': [['key', 'val'], ['other_key', 'other_val']],
+        'project_blacklist': [],
+        'tag_blacklist': [],
+        'zone': 'zone1',
+    }
+
+
+@pytest.fixture
+def authority(mocker, monkeypatch, authority_config, auth_client):
+    monkeypatch.setattr(
+        'gordon_janitor_gcp.gce_authority.auth.GoogleAuthClient',
+        mocker.Mock(return_value=auth_client))
+    rrset_channel = asyncio.Queue()
+    authority = gce_authority.GCEAuthority(
+        authority_config, rrset_channel)
+    authority.session = auth_client._session
+    return authority
+
+
+@pytest.fixture
+def get_mock_client(mocker, get_gce_client):
+    def _create_fake(klass, *args, **kwargs):
+        client = get_gce_client(klass, *args, **kwargs)
+        fake_client = mocker.Mock(client)
+        return fake_client
+    return _create_fake
+
+
+@pytest.mark.asyncio
+async def test_builder_creates_proper_authority(mocker, authority_config):
+    gce_client = mocker.MagicMock(name='gce_client')
+    crm_client = mocker.MagicMock(name='crm_client')
+    auth_client = mocker.MagicMock(name='auth_client')
+    authority_mod_patch = 'gordon_janitor_gcp.gce_authority.'
+    mocker.patch(authority_mod_patch + 'gcp_clients.GCRMClient', crm_client)
+    mocker.patch(authority_mod_patch + 'gcp_clients.GCEClient', gce_client)
+    mocker.patch(authority_mod_patch + 'auth.GoogleAuthClient', auth_client)
+
+    rrset_channel = 'channel'
+    kwargs = {'other': 'kwargs'}
+    builder = gce_authority.GCEAuthorityBuilder(
+        authority_config, rrset_channel, **kwargs)
+    session = builder.session
```

Comment:
Do you not need to yield for this? I'd expect aiohttp to complain about a session not being closed.

Reply:
I'm just "grabbing" the attribute of the builder and assigning it to its own variable for clarity so I don't have to yield anything. The next line (`with contextlib.closing(session))`) takes care of closing the session.

Reply:
hm wonder how I missed that...